tests/check-core.c

```diff
@@ -522,6 +523,66 @@ START_TEST(bootman_timeout_test)
 }
 END_TEST
 
+START_TEST(bootman_writer_simple_test)
+{
+        autofree(CbmWriter) *writer = CBM_WRITER_INIT;
+
+        fail_if(!cbm_writer_open(writer), "Failed to create writer");
+
+        cbm_writer_append(writer, "Bob");
+        cbm_writer_append(writer, "-");
+        cbm_writer_append(writer, "Jim");
+
+        fail_if(cbm_writer_error(writer) != 0, "Error should be 0");
+
+        cbm_writer_close(writer);
+        fail_if(!writer->buffer, "Failed to get writer data");
+        fail_if(!streq(writer->buffer, "Bob-Jim"), "Returned data is incorrect");
+}
+END_TEST
+
+START_TEST(bootman_writer_printf_test)
+{
+        autofree(CbmWriter) *writer = CBM_WRITER_INIT;
+
+        fail_if(!cbm_writer_open(writer), "Failed to create writer");
+
+        cbm_writer_append_printf(writer, "%s = %d", "Jim", 12);
+        fail_if(cbm_writer_error(writer) != 0, "Error should be 0");
+
+        cbm_writer_close(writer);
+        fail_if(!writer->buffer, "Failed to get writer data");
+        fail_if(!streq(writer->buffer, "Jim = 12"), "Returned data is incorrect");
+}
+END_TEST
+
+START_TEST(bootman_writer_mut_test)
+{
+        autofree(CbmWriter) *writer = CBM_WRITER_INIT;
+        char *data = NULL;
+
+        fail_if(!cbm_writer_open(writer), "Failed to create writer");
+
+        cbm_writer_append(writer, "One");
+        cbm_writer_append(writer, "Two");
+        cbm_writer_close(writer);
+        data = writer->buffer;
+        fail_if(!data, "Failed to get data");
+
+        /* Should not cause segfaults! */
+        cbm_writer_append(writer, "Three");
+        fail_if(cbm_writer_error(writer) == 0, "Error should be zero");
```

Comment:
OK so we only use stdlib errors, and set enomem on an open_memstream failed, and return in all other instances without setting errors?